src/PKSim.Core/Commands/EditParameterCommand.cs

```diff
@@ -104,6 +111,64 @@ private void setBuildingBlockAlteredFlag(IExecutionContext context)
          }
       }
 
+      protected void SaveValueOriginFor(IParameter parameter)
+      {
+         _oldValueOrigin = parameter.ValueOrigin.Clone();
+      }
+
+      protected virtual void UpdateParameter(IExecutionContext context)
+      {
+         var bbParameter = OriginParameterFor(_parameter, context);
+         UpdateParameter(_parameter, context);
+         UpdateParameter(bbParameter, context);
+         UpdateValueOriginInParameters(_parameter, bbParameter);
+      }
+
+      protected virtual void UpdateValueOriginInParameters(IParameter parameter, IParameter buildingBlockParameter)
+      {
+         //This is an undo command
+         if (_oldValueOrigin != null)
+         {
+            //Simply reset the parameter value origin
+            var currentValueOrigin = parameter.ValueOrigin.Clone();
+            updateValueOriginIn(parameter, _oldValueOrigin);
+            updateValueOriginIn(buildingBlockParameter, _oldValueOrigin);
+            _oldValueOrigin = currentValueOrigin;
+            return;
+         }
+
+         //This is a value being set directly: Only save the ValueOrigin for default parameter
+         if (!parameter.ValueOrigin.Default)
+            return;
+
+         SaveValueOriginFor(parameter);
+         markDefaultValueOriginAsChanged(parameter.ValueOrigin);
+      }
+
+      private static bool valueOriginShouldBeUpdatedAutomatically(ValueOrigin valueOrigin)
+      {
+         return valueOrigin.Source == ValueOriginSources.Undefined &&
```

Comment:
To be discussed. This will only be updated if both source and method are undefined. Otherwise, the user changed something already so this should not be resetted